src/parser/ast/mod.rs

```diff
@@ -0,0 +1,29 @@
+use core::fmt::Debug;
+
+pub mod expression;
+pub mod statement;
+pub mod tl;
+
+#[derive(Clone, Debug)]
+pub struct DebugSymRef {
+    pub start: usize,
+    pub end: usize,
+}
```

Comment:
Is this a TODO?  At this stage I think it's OK to have unfinished parts, but I will request that at least there are some comments with TODO, with a small explanation of what's the intention of the types.

Please, add as many TODOs as necessary to this PR!

Reply:
DebugSymRef will grow with more data as we add more information to errors and debug information, currently it is a placeholder, that such info should be in each node of the AST.

Everything is a big TODO, but I will add a TODO for this.

Reply:
TODO added